### PR TITLE
Fixes #15 deploys master branch to google cloud

### LINKED ISSUE #15: Deploy Master Branch automatically to Github
In order to get this project working we need to establish a DevOps flow. Apart from getting a Heroku deployment of the development branch, the goal is to have a continuous Master branch deployment to the Google cloud with Kubernetes. Please implement this.

### PR DESCRIPTION
**Pull request by nikhilrayaprolu**
fixes #15 
Made all changes required for the deployment and got the container working on google cloud, I have tested this on my own repo, and got it working. related travis report is at https://travis-ci.org/nikhilrayaprolu/yacy_grid_mcp .

One needs admin account to add authentication keys google cloud.
image

the deployment address is at http://130.211.167.207:8100/

### CODE REVIEW

**Review by niranjan94 — CHANGES_REQUESTED** [review]
.

File: deploy_staging.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+echo $GCLOUD_SERVICE | base64 --decode -i > ${HOME}/gcloud-service-key.json
+cat ${HOME}/gcloud-service-key.json
+gcloud auth activate-service-account --key-file ${HOME}/gcloud-service-key.json
+
+gcloud --quiet config set project $PROJECT_NAME_STG
+gcloud --quiet config set container/cluster $CLUSTER_NAME_STG
+gcloud --quiet config set compute/zone ${CLOUDSDK_COMPUTE_ZONE}
+gcloud --quiet container clusters get-credentials $CLUSTER_NAME_STG
+
+gcloud docker -- push gcr.io/${PROJECT_NAME_STG}/${DOCKER_IMAGE_NAME}
```
Comment (niranjan94) [inline deploy_staging.sh]: Why do we need to use google cloud's private repository hosting service ? This is anyway an open source project. So, a private container hosting service doesn't make sense and we'll be paying extra for it. (Google cloud platform isn't cheap) 

We can use Docker Hub instead (https://hub.docker.com) like how we do for Open Event. 

https://hub.docker.com/u/eventyay/

**Review by niranjan94 — CHANGES_REQUESTED** [review]
@nikhilrayaprolu @mariobehling

File: deploy_staging.sh (hunk 1 above)
Comment (nikhilrayaprolu) [reply deploy_staging.sh]: @niranjan94 as far as I know docker hub is just a docker registry and not a container service.so we could just push and test images on docker hub, but cannot deploy or host them.

Comment (mariobehling) [reply deploy_staging.sh]: Yes, public is better. I am checking if @Orbiter has access to the Yacy org on Docker. If not we need to set up another org.

Comment (mariobehling) [reply deploy_staging.sh]: @nikhilrayaprolu Please check out the flow to go through Dockerhub in the meantime.

Comment (niranjan94) [reply deploy_staging.sh]: @nikhilrayaprolu you can host the docker images there and use them in your kubernetes configuration. A replacement for `gcr.io`. 

We have been doing that for quite sometime with eventyay. https://hub.docker.com/u/eventyay/

Comment (nikhilrayaprolu) [reply deploy_staging.sh]: @niranjan94 do you mean to replace just gcr.io with dockerhub for image hosting and run the container on hosting services like google cloud?

Comment (niranjan94) [reply deploy_staging.sh]: @nikhilrayaprolu yes something like that ... See

- https://github.com/fossasia/open-event-orga-server/blob/nextgen/kubernetes/travis/deploy-nextgen.sh#L33
- https://github.com/fossasia/open-event-orga-server/blob/nextgen/kubernetes/yamls/api-server/api-deployment.yml#L21
- https://hub.docker.com/r/eventyay/nextgen-api-server/tags/

Comment (nikhilrayaprolu) [reply deploy_staging.sh]: @niranjan94 I found these files where eventyay related repositories, where dockerhub is used for image registry and google cloud for hosting. https://github.com/fossasia/open-event-frontend/blob/development/kubernetes/travis/deploy.sh . do you mean to proceed like this?

**Comment by nikhilrayaprolu** [comment]
@mariobehling I have replaced the google cloud registry with docker registry. @Orbiter I don't have permissions to add service keys and docker related keys into travis of this repository.
right now the image for yacygrid is available at https://hub.docker.com/r/nikhilrayaprolu/yacygridmcp/

**Comment by nikhilrayaprolu** [comment]
I have tested deployment with my forked repo and it got passed, the related travis log is at https://travis-ci.org/nikhilrayaprolu/yacy_grid_mcp/builds/254743090?utm_source=github_status&utm_medium=notification

**Comment by mariobehling** [comment]
@nikhilrayaprolu Ok, let's merge this and iterate from here.

**Comment by nikhilrayaprolu** [comment]
@mariobehling can I get temporary access to admin rights for this repository so that I could add the keys into travis.

**Pull request merged**